src/boot/efi/boot.c

```diff
@@ -2458,10 +2549,22 @@ static EFI_STATUS image_start(
                 const char *extra = smbios_find_oem_string("io.systemd.boot.kernel-cmdline-extra");
                 if (extra) {
                         _cleanup_free_ char16_t *tmp = TAKE_PTR(options), *extra16 = xstr8_to_16(extra);
-                        options = xasprintf("%ls %ls", tmp, extra16);
+                        if (isempty(tmp))
+                                options = TAKE_PTR(extra16);
+                        else
+                                options = xasprintf("%ls %ls", tmp, extra16);
                 }
         }
 
+        /* Prefix profile if it's non-zero */
+        if (entry->profile > 0) {
+                _cleanup_free_ char16_t *tmp = TAKE_PTR(options);
+                if (isempty(tmp))
+                        options = xasprintf("@%u", entry->profile);
+                else
+                        options = xasprintf("@%u %ls", entry->profile, tmp);
+        }
```

Comment:
Frankly I don't like using the cmdline as a vehicle for passing through the profile. Why not pass it as an argument? Didn't you add code for that as well? That seems much cleaner than this.

Reply:
Can't we use `EFI_SHELL_PARAMETERS_PROTOCOL` to pass `argc/argv` to stub instead? And then maybe write a dead simple CLI parser in stub so the profile can be passed via `--profile`?

Reply:
> What do you mean by "as an argument"?

`argc` and `argv`